up-subscription/src/subscription_manager.rs

```diff
@@ -147,6 +147,11 @@ pub(crate) async fn handle_message(
                     topic,
                     respond_to,
                 } => {
+                    let mut subscr_before_ins = 0;
+                    if let Some(subscriber_set) = topic_subscribers.get(&topic) {
```

Comment:
For this, I'd suggest to make use of the HashMap::insert() return value: 

```
let is_new = topic_subscribers
                        .entry(topic.clone())
                        .or_default()
                        .insert(subscriber);
```
[...]

`if is_new {`

This would get rid of the `subscribers_count` thing entirely - think that would work?

Reply:
Nice, that looks much better!

https://github.com/jjj-vtm/up-subscription-rust/blob/3c350f97035ea7f5e9c7fba2c3696d94e0cde2ed/up-subscription/src/subscription_manager.rs#L152C21-L168C65